models/aeif_psc_alpha.cpp

```diff
@@ -0,0 +1,538 @@
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+nest::RecordablesMap< nest::aeif_psc_alpha >
+  nest::aeif_psc_alpha::recordablesMap_;
+
+namespace nest // template specialization must be placed in namespace
+{
+// Override the create() method with one call to RecordablesMap::insert_()
+// for each quantity to be recorded.
+template <>
+void
+RecordablesMap< aeif_psc_alpha >::create()
+{
+  // use standard names whereever you can for consistency!
+  insert_(
+    names::V_m, &aeif_psc_alpha::get_y_elem_< aeif_psc_alpha::State_::V_M > );
+  insert_( names::I_syn_ex,
+    &aeif_psc_alpha::get_y_elem_< aeif_psc_alpha::State_::I_EXC > );
+  insert_( names::I_syn_in,
+    &aeif_psc_alpha::get_y_elem_< aeif_psc_alpha::State_::I_INH > );
+  insert_(
+    names::w, &aeif_psc_alpha::get_y_elem_< aeif_psc_alpha::State_::W > );
+}
+}
+
+extern "C" int
+nest::aeif_psc_alpha_dynamics( double,
+  const double y[],
+  double f[],
+  void* pnode )
+{
+  // a shorthand
+  typedef nest::aeif_psc_alpha::State_ S;
+
+  // get access to node so we can almost work as in a member function
+  assert( pnode );
+  const nest::aeif_psc_alpha& node =
+    *( reinterpret_cast< nest::aeif_psc_alpha* >( pnode ) );
+
+  // y[] here is---and must be---the state vector supplied by the integrator,
+  // not the state vector in the node, node.S_.y[].
+
+  // The following code is verbose for the sake of clarity. We assume that a
+  // good compiler will optimize the verbosity away ...
+
+  // shorthand for state variables
+  const double& V =
+    std::min( y[ S::V_M ], node.P_.V_peak_ ); // enforce upper limit on V_m
+  const double& dI_syn_ex = y[ S::DI_EXC ];
+  const double& I_syn_ex = y[ S::I_EXC ];
+  const double& dI_syn_in = y[ S::DI_INH ];
+  const double& I_syn_in = y[ S::I_INH ];
+  const double& w = y[ S::W ];
+
+  const double I_spike = node.P_.Delta_T == 0.
+    ? 0.
+    : ( node.P_.g_L * node.P_.Delta_T
+        * std::exp( ( V - node.P_.V_th ) / node.P_.Delta_T ) );
+
+  // dv/dt
+  f[ S::V_M ] =
+    ( -node.P_.g_L * ( V - node.P_.E_L ) + I_spike + I_syn_ex - I_syn_in - w
+      + node.P_.I_e + node.B_.I_stim_ ) / node.P_.C_m;
+
+  f[ S::DI_EXC ] = -dI_syn_ex / node.P_.tau_syn_ex;
+  // Exc. synaptic current (pA)
+  f[ S::I_EXC ] = dI_syn_ex - I_syn_ex / node.P_.tau_syn_ex;
+
+  f[ S::DI_INH ] = -dI_syn_in / node.P_.tau_syn_in;
+  // Inh. synaptic current (pA)
+  f[ S::I_INH ] = dI_syn_in - I_syn_in / node.P_.tau_syn_in;
+
+  // Adaptation current w.
+  f[ S::W ] = ( node.P_.a * ( V - node.P_.E_L ) - w ) / node.P_.tau_w;
+
+  return GSL_SUCCESS;
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::aeif_psc_alpha::Parameters_::Parameters_()
+  : V_peak_( 0.0 )    // mV, should not be larger that V_th+10
```

Comment:
I think the idea here is that the larger the V_th to V_peak is, the worse does the exploding-exponential problem become. But I think we handle this well elsewhere now, so the comment can be removed.